### DISCUSSION TITLE
Zero Copy/Data sharing use case

### DISCUSSION THREAD

**Discussion by ghost** [eProsima/Fast-DDS#5072] [Q&A]
Hi all,
 As part of data transfer performance improvement & to remove _Serializations-Deserializations_ during data transfer (ref. [https://github.com/eProsima/Fast-DDS/discussions/3663](url)), I had configured data sharing mode in ROS2 with FastDDS MW using XML file (as documented in [https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/zero_copy/zero_copy.html#use-case-zero-copy](url)) & no performance improvement was found.

So, it will be of great help if anyone can provide some light on the below queries:

1. When should the Data sharing mode to be used?
2. Is the zero copy applicable between processes running on different machines?
3. For processes on same machine, shared memory is the default transport. Is the shared memory transport technique same as zero copy/loan mechanism?
4. Is there any other optimization mechanisms available in Fast-DDS?

Thanks in advance..!

**Comment by Mario-DL** [comment]
Hi @arunjs-dev,

The use of Zero Copy in ROS 2 can be achieved by following the instructions in [rmw_fastrtps](https://github.com/eProsima/rmw_fastrtps?tab=readme-ov-file#enable-zero-copy-data-sharing). 
Zero Copy can either be used with intraprocess or datasharing delivery kinds if the requirements are met and it is not applicable among different hosts.

For any further support you might consider contacting [eProsima's commercial support team](https://github.com/eProsima/Fast-DDS/tree/master#getting-help) if this is a need for your project.

**Reply by ghost** [reply]
Hi @Mario-DL ,
       As mentioned, I had configured zero copy by loading the XML specified in [rmw_fastrtps](https://github.com/eProsima/rmw_fastrtps?tab=readme-ov-file#enable-zero-copy-data-sharing). But I'm getting an error message during ros2 application execution specifying that the unbounded types are not supported, eventhough I'm using plain data types mentioned in [constraints](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/zero_copy/zero_copy.html#constraints).
